erpnext/asset/doctype/asset/asset.js

```diff
@@ -0,0 +1,304 @@
+// Copyright (c) 2016, Frappe Technologies Pvt. Ltd. and contributors
+// For license information, please see license.txt
+
+frappe.provide("erpnext.asset");
+
+frappe.ui.form.on('Asset', {
+	onload: function(frm) {
+		frm.set_query("item_code", function() {
+			return {
+				"filters": {
+					"disabled": 0,
+					"is_fixed_asset": 1,
+					"is_stock_item": 0
+				}
+			};
+		});
+
+		frm.set_query("warehouse", function() {
+			return {
+				"filters": {
+					"company": frm.doc.company,
+					"is_group": 0
+				}
+			};
+		});
+	},
+
+	refresh: function(frm) {
+		frappe.ui.form.trigger("Asset", "is_existing_asset");
+		frm.toggle_display("next_depreciation_date", frm.doc.docstatus < 1);
+		frm.events.make_schedules_editable(frm);
+
+		if (frm.doc.docstatus==1) {
+			if (in_list(["Submitted", "Partially Depreciated", "Fully Depreciated"], frm.doc.status)) {
+				frm.add_custom_button("Transfer Asset", function() {
+					erpnext.asset.transfer_asset(frm);
+				});
+
+				frm.add_custom_button("Scrap Asset", function() {
+					erpnext.asset.scrap_asset(frm);
+				});
+
+				frm.add_custom_button("Sell Asset", function() {
+					erpnext.asset.make_sales_invoice(frm);
+				});
+
+			} else if (frm.doc.status=='Scrapped') {
+				frm.add_custom_button("Restore Asset", function() {
+					erpnext.asset.restore_asset(frm);
+				});
+			}
+			if (frm.doc.status=='Submitted' && !frm.doc.is_existing_asset && !frm.doc.purchase_invoice) {
+				frm.add_custom_button(__("Purchase Invoice"), function() {
+					frm.trigger("make_purchase_invoice");
+				}, __("Make"));
+			}
+			if (frm.doc.maintenance_required && !frm.doc.maintenance_schedule) {
+				frm.add_custom_button(__("Asset Maintenance"), function() {
+					frm.trigger("create_asset_maintenance");
+				}, __("Make"));
+			}
+			frm.page.set_inner_btn_group_as_primary(__("Make"));
+			frm.trigger("setup_chart");
+		}
+	},
+
+	setup_chart: function(frm) {
+		var x_intervals = [frm.doc.purchase_date];
+		var asset_values = [frm.doc.gross_purchase_amount];
+		var last_depreciation_date = frm.doc.purchase_date;
+
+		if(frm.doc.opening_accumulated_depreciation) {
+			last_depreciation_date = frappe.datetime.add_months(frm.doc.next_depreciation_date,
+				-1*frm.doc.frequency_of_depreciation);
+
+			x_intervals.push(last_depreciation_date);
+			asset_values.push(flt(frm.doc.gross_purchase_amount) -
+				flt(frm.doc.opening_accumulated_depreciation));
+		}
+
+		$.each(frm.doc.schedules || [], function(i, v) {
```

Comment:
(i, v) is the default example but if you use for v something like `schedule` as v its more readable for later maintaince.